### PR TITLE
CB-10991 Allow to edit on Azure and GCP.

### PR DESCRIPTION
**Pull request by topolyai5**
See detailed description in the commit message.

### CODE REVIEW

File: environment/src/main/java/com/sequenceiq/environment/environment/dto/AuthenticationDtoConverter.java [hunk 1]
```diff
@@ -21,12 +21,9 @@ public AuthenticationDtoConverter(PublicKeyValidator publicKeyValidator) {
 
     public EnvironmentAuthentication dtoToAuthentication(AuthenticationDto authenticationDto) {
         EnvironmentAuthentication environmentAuthentication = new EnvironmentAuthentication();
-        if (StringUtils.isNotEmpty(authenticationDto.getPublicKey())) {
-            ValidationResult validationResult = publicKeyValidator.validatePublicKey(authenticationDto.getPublicKey());
-            if (!validationResult.hasError()) {
-                List<String> parts = Arrays.asList(StringUtils.split(authenticationDto.getPublicKey(), " "));
-                environmentAuthentication.setPublicKey(String.format("%s %s %s", parts.get(0), parts.get(1), authenticationDto.getLoginUserName()));
-            }
+        if (isValidSshKey(authenticationDto.getPublicKey())) {
```
Comment (doktoric) [inline environment/src/main/java/com/sequenceiq/environment/environment/dto/AuthenticationDtoConverter.java:24]: dont we need to check first that the public key exists?

File: environment/src/main/resources/application.yml [hunk 2]
```diff
@@ -101,6 +101,16 @@ environment:
   enabledplatforms: AWS,AZURE,YARN,GCP
   enabledParentPlatforms: AWS,MOCK
   enabledChildPlatforms: YARN,MOCK
+  existing:
+    ssh:
+      key:
+        update:
+          support: AWS,MOCK
```
Comment (doktoric) [inline environment/src/main/resources/application.yml:108]: ycloud ?

Comment (doktoric) [reply environment/src/main/resources/application.yml:108]: what happens with if we add a new provider but that is not here

Comment (topolyai5) [reply environment/src/main/resources/application.yml:108]: We won't update the raw public key or public key id.

File: environment/src/main/java/com/sequenceiq/environment/environment/dto/AuthenticationDtoConverter.java (hunk 1 above)
Comment (topolyai5) [reply environment/src/main/java/com/sequenceiq/environment/environment/dto/AuthenticationDtoConverter.java:24]: it is the raw public key. We want to validate the format here. However, I moved the conditions onto a method. The conditions are the same than previously.

**Review by doktoric — APPROVED** [review]

**Pull request merged**